### PR TITLE
Store: Add tracks events for product category management

### PR DESCRIPTION
**Pull request by justinshreve**
This PR adds tracks events to product category management. See #20295.

Open your console and paste `localStorage.setItem('debug', '*analytics*');`. You will be looking for `woocommerce:analytics track` events.

Please test the following events and make sure you see them in your console:

* Searching from the product category listing page.
* Save/create a new product category.
* Save/update a product category.
* Delete a product category.

### CODE REVIEW

**Comment by dmsnell** [comment]
I'd encourage a pattern here where we hide the analytics calls from within `mapDispatchToProps` so that the React component itself doesn't need to concern itself with both the _actions_ themselves _and_ the analytics calls. It can leave a simpler component with fewer testing issues and can leave those analytics (where possible, which is most but not all calls) as incidental details of the more important API that the component presents. For example, if we are tracking a metric when creating a category, then our component can continue to just call `this.props.createCategory()` and we can make sure in `connect()` / `mapDispatchToProps` that the analytics is called.

In general Calypso work we can take advantage of the `withAnalytics()` wrapper to make this easy… (need to verify this already exists too in woo)

```js
const mapDispatchToProps = ( {
	createCategory: withAnalytics(
		recordTracksEvent( 'created_a_category' ),
		createCategory,
	),
	…
} )
```

**Comment by justinshreve** [comment]
Thanks for the feedback! I definitely like that pattern. I gave it a try and it seems to not be working correctly for our extension. I keep getting a warning about dispatch being undefined. 

We are on our support week this week, so I'd like to keep this PR the way it is for now, so that we can start tracking things out the gate when we launch next week, but we can look into getting `withAnalytics` working and moving our tracks to that pattern.

**Comment by dmsnell** [comment]
If `dispatch` is undefined it just means we have to define it.

```js
const mapDispatchToProps = dispatch => ( {
	createCategory: () => dispatch( withAnalytics(
		recordTracksEvent( 'created_a_category' ),
		createCategory,
	) ),
} )
```

It's a good pattern to use, or at least I'd strongly encourage us not to make analytics calls directly in React components in _new code_ unless we have a good reason not to. It's also a trivial change to the PR :wink:

File: client/extensions/woocommerce/app/product-categories/index.js [hunk 1]
```diff
@@ -132,13 +133,12 @@ function mapStateToProps( state ) {
...
-		},
-		dispatch
-	);
-}
+const mapDispatchToProps = dispatch => ( {
+	searchProductCategories: ( siteId, query ) => withAnalytics(
+		recordTrack( 'calypso_woocommerce_product_category_search', query ),
+		fetchProductCategories( siteId, query )( dispatch ),
+	),
+	fetchProductCategories: ( ...args ) => fetchProductCategories( ...args )( dispatch ),
```
[the remark this answers is not in the document]
Comment (justinshreve) [reply client/extensions/woocommerce/app/product-categories/index.js]: Yes. I would like to update fetchProductCatrgories to just return an action, but would like to get these events in, and handle action changes in a separate PR after our support week is over.

**Pull request merged**